shell/platform/darwin/ios/ios_surface_metal_impeller.mm

```diff
@@ -25,7 +26,11 @@
 }
 
 // |IOSSurface|
-IOSSurfaceMetalImpeller::~IOSSurfaceMetalImpeller() = default;
+IOSSurfaceMetalImpeller::~IOSSurfaceMetalImpeller() {
+  if ([layer_ isKindOfClass:[FlutterMetalLayer class]]) {
+    [(FlutterMetalLayer*)layer_.get() invalidate];
```

Comment:
I have a preference for @knopp s patch. It really brittle for `IOSSurfaceMetalImpeller` to know to check if the layers is of a specific type to call a method on it.